[SPARK-5654] Integrate SparkR into Apache Spark

davies opened the pull request:
This PR integrate SparkR-pkg [1] in Spark, it include almost complete API for Spark core and SparkSQL.

The following things could be done later (out of this PR):
1. generating documentation
2. R lint
3. API audit
4. more tests

Known issues:
1. does not work in YARN cluster mode

[1] github.com/amplab-extras/SparkR-pkg

evertlammerts committed:
fixed parquetFile signature

evertlammerts committed:
added tests

shivaram committed:
Merge pull request #184 from davies/socket

[SPARKR-155] use socket in R worker

Commit:
Merge branch 'sparkr-sql' into dev

davies committed:
Merge branch 'master' of github.com:amplab-extras/SparkR-pkg into random

davies committed:
send back the port via temporary file

Commit:
Three more DataFrame methods

- `repartition`
- `distinct`
- `sampleDF`

Commit:
Fix docs and indents

lythesia committed:
modify tests for repartition

Commit:
numToInt utility function

Added `numToInt` converter function for allowing numeric arguments when integers are required. Updated `repartition`.

lythesia committed:
keep partitions check

Commit:
Update `sampleDF` test

lythesia committed:
minor fixes

shivaram committed:
Merge pull request #187 from cafreeman/sparkr-sql

Three more DataFrame methods

davies committed:
address comment

lythesia committed:
fix indents

shivaram committed:
Merge pull request #177 from lythesia/master

[SPARKR-152] Support functions to change number of RDD partitions (coalesce, repartition)

davies committed:
wait atmost 100 seconds

shivaram committed:
Set Spark version 1.3.0 in Windows build

davies committed:
add Column and expression

davies committed:
Merge branch 'sparkr-sql' of github.com:amplab-extras/SparkR-pkg into column

Conflicts:
	pkg/R/utils.R
	pkg/inst/tests/test_sparkSQL.R

hlin09 committed:
Fix comments.

hlin09 committed:
Fix comments.

hlin09 committed:
Add tests for broadcast variables.

shivaram committed:
Merge pull request #186 from hlin09/funcDep3

[SPARKR-142][SPARKR-196] (Step 2) Replaces getDependencies() with cleanClosure to capture UDF closures and serialize them to worker.

shivaram committed:
Fix stderr redirection on executors

shivaram committed:
Remove unused function

Commit:
Update SparkRBackend.scala

shivaram committed:
Merge pull request #189 from shivaram/stdErrFix

Fix stderr redirection on executors

shivaram committed:
Merge pull request #190 from shivaram/SPARKR-79

[SPARKR-79] Remove unused function

concretevitamin committed:
Merge pull request #178 from davies/random

[SPARKR-204] use random port in backend

felixcheung committed:
Doc updates: build and running on YARN

davies committed:
use S4 methods

davies committed:
fix cyclic namespace dependency

davies committed:
fix approxCountDistinct

davies committed:
refactor

davies committed:
fix signature

davies committed:
fix first(0

Commit:
define generic for 'first' in RDD API

Commit:
Fix brackets

Commit:
return object instead of a list of one object

shivaram committed:
Merge pull request #192 from cafreeman/sparkr-sql

define generic for 'first' in RDD API

Commit:
Updates as per feedback on sparkR-submit

Commit:
Update DataFrame.R

Commit:
Update column.R

davies committed:
Merge branch 'sparkr-sql' of github.com:amplab-extras/SparkR-pkg into column

shivaram committed:
Merge pull request #188 from davies/column

[SPARKR-189] [SPARKR-190] Column and expression

davies committed:
launch R worker by a daemon

davies committed:
Merge branch 'sparkr-sql' of github.com:amplab-extras/SparkR-pkg into api

Conflicts:
	pkg/R/RDD.R

Commit:
Fixed small typos

davies committed:
fix test and docs

davies committed:
Merge branch 'sparkr-sql' of github.com:amplab-extras/SparkR-pkg into group

Conflicts:
	pkg/NAMESPACE
	pkg/R/DataFrame.R
	pkg/R/utils.R
	pkg/inst/tests/test_sparkSQL.R

davies committed:
Merge branch 'group' of github.com:davies/SparkR-pkg into group

Conflicts:
	pkg/NAMESPACE
	pkg/R/DataFrame.R

davies committed:
address comments

Commit:
New 1.3 repo and updates to `column.R`

Updated column to use `functions` instead of `Dsl` in accordance with the new API changes.

Also created separate classes for `asc` and `desc`.

davies committed:
shutdown the JVM after R process die

shivaram committed:
Merge pull request #195 from cafreeman/sparkr-sql

New 1.3 repo and updates to `column.R`

Commit:
Refactor `join` generic for use with `DataFrame`

Commit:
`join`, `sort`, and `filter`

New DataFrame methods:

- `join`
- `sort`
- `orderBy`
- `filter`
- `where`

davies committed:
Merge branch 'sparkr-sql' of github.com:amplab-extras/SparkR-pkg into api

Conflicts:
	pkg/src/build.sbt

davies committed:
fix tests

Commit:
small update on yarn deploy mode.

Commit:
Fixed indent in `join` test.

Commit:
`selectExpr`

Commit:
update export

shivaram committed:
Merge pull request #197 from cafreeman/sparkr-sql

[SparkR-209] `join`, `sort`, `filter` methods for DataFrame

Commit:
Merge branch 'dfMethods' into sparkr-sql

Commit:
remove extra line

Commit:
Fix test

davies committed:
Merge branch 'sparkr-sql' of github.com:amplab-extras/SparkR-pkg into load

Conflicts:
	pkg/NAMESPACE
	pkg/R/utils.R
	pkg/inst/tests/test_sparkSQL.R
	pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/SQLUtils.scala

shivaram committed:
Merge pull request #191 from felixcheung/doc

Doc updates: build and running on YARN

davies committed:
add HiveContext (commented)

shivaram committed:
Merge pull request #194 from davies/api

[SPARKR-210] Some APIs in SQLContext

Commit:
Update SparkRBackend.scala

shivaram committed:
Remove SparkR snapshot build
We now have 1.3.0 RC2 on Apache Staging

Commit:
Update sparkR.R

Commit:
[SPARKR-156] phase 1: implement zipWithUniqueId() of the RDD class.

Commit:
Fixed docs

Commit:
Merge branch 'sparkr-sql' of https://github.com/amplab-extras/SparkR-pkg into sparkr-sql

shivaram committed:
Use rc2 explicitly

shivaram committed:
Merge pull request #200 from shivaram/sparkr-sql-build

Revert SparkR snapshot dependency

shivaram committed:
Merge pull request #198 from cafreeman/sparkr-sql

`selectExpr`

Commit:
clean up docs

Commit:
'explain`

Commit:
`toJSON`

Commit:
`isLocal`

Commit:
`showDF`

Commit:
fix test

Commit:
'saveAsParquetFile`

davies committed:
address comments

davies committed:
support Column expression in agg()

davies committed:
Merge branch 'sparkr-sql' of github.com:amplab-extras/SparkR-pkg into load

Conflicts:
	pkg/NAMESPACE
	pkg/R/SQLContext.R

Commit:
Update `explain` and fixed doc for `toJSON`

shivaram committed:
Merge pull request #202 from cafreeman/sparkr-sql

[SparkR-209] `explain`, `toJSON`, `isLocal`, `showDF`, `saveAsParquetFile` DataFrame methods

davies committed:
fix style

davies committed:
Merge branch 'sparkr-sql' of github.com:amplab-extras/SparkR-pkg into group

Conflicts:
	pkg/NAMESPACE
	pkg/R/DataFrame.R
	pkg/inst/tests/test_sparkSQL.R

davies committed:
Merge branch 'sparkr-sql' of github.com:amplab-extras/SparkR-pkg into load

Conflicts:
	pkg/NAMESPACE

davies committed:
fix HiveContext

shivaram committed:
Merge pull request #199 from davies/load

[SPARKR-203] load and save DataFrame

davies committed:
fix tests

shivaram committed:
Fix HiveContext package name

shivaram committed:
Fix test hive context package name

shivaram committed:
Merge pull request #203 from shivaram/sparkr-hive-fix

Fix HiveContext package name

davies committed:
close monitor connection in sparkR.stop()

davies committed:
Merge branch 'die' of github.com:davies/SparkR-pkg into die

Commit:
Use mapply() instead of for statement.

davies committed:
address comments

shivaram committed:
Merge pull request #201 from sun-rui/SPARKR-156_1

[SPARKR-156] phase 1: implement zipWithUniqueId() of the RDD class.

davies committed:
add test to start and stop context multiple times

Commit:
`dropTempTable`

Add `dropTempTable` function and update tests to drop the test table at the end of every test.

Commit:
`insertInto`

Commit:
`intersect`, `subtract`, `unionAll`

davies committed:
remove test, catch exception

davies committed:
improve logging

Commit:
Update SparkRBackendHandler.scala

shivaram committed:
Merge pull request #180 from davies/group

[SPARKR-191] groupBy and agg() API for DataFrame

Commit:
New `select` method

Added another version of `select` that will take a list as an argument (instead of just specific column names or expression)

Commit:
`withColumn` and `withColumnRenamed`

Commit:
Merge branch 'dfMethods' into sparkr-sql

Commit:
[SPARKR-156] phase 2: implement zipWithIndex() of the RDD class.

Commit:
Fix indentation.

shivaram committed:
Remove print statement in SparkRBackendHandler
This print statement is noisy for SQL methods which have multiple APIs
(like loadDF). We already have a better error message when no valid
methods are found

Commit:
Fix minor typos in the function description.

shivaram committed:
Merge pull request #206 from sun-rui/SPARKR-156_2

[SPARKR-156] phase 2: implement zipWithIndex() of the RDD class.

Commit:
update tests to guarantee row order

shivaram committed:
Merge pull request #207 from shivaram/err-remove

Remove print statement in SparkRBackendHandler

Commit:
Merge branch 'sparkr-sql' of https://github.com/amplab-extras/SparkR-pkg into sparkr-sql

davies committed:
use fork package

hlin09 committed:
Add checks for namespace access operators in cleanClosure.

Commit:
Fix example for `dropTempTable`

shivaram committed:
Merge pull request #204 from cafreeman/sparkr-sql

[SparkR-209] Remaining DataFrame methods + `dropTempTable`

hlin09 committed:
Add a test for access operators.

hlin09 committed:
Move setGeneric() to 00-generics.R.

hlin09 committed:
Move getJRDD and broadcast's value to 00-generic.R.

shivaram committed:
Merge pull request #210 from hlin09/hlin09

Add checks for namespace access operators (:: and :::) in cleanClosure.

hlin09 committed:
Move roxygen doc back to implementation.

davies committed:
switch back to use parallel

davies committed:
make RStudio happy

hlin09 committed:
Remove duplication.

hlin09 committed:
Remove missingOrInteger.

davies committed:
add config for using daemon, refactor

davies committed:
retry

davies committed:
readFully()

shivaram committed:
Merge pull request #211 from hlin09/generics

[SPARKR-220] Move setGeneric() to a separate file (00-generics.R) to make S4 generic methods easier to follow, and visible to all data types' implementation.

shivaram committed:
Merge pull request #196 from davies/die

[SPARKR-198] shutdown the JVM after R process die

davies committed:
createDataFrame from rdd

davies committed:
support date.frame and Date/Time

davies committed:
remove dependency of jsonlite

davies committed:
improve comment and logging

davies committed:
include grouping columns in agg()

add docs for groupBy() and agg()

davies committed:
add more selecter for DataFrame

shivaram committed:
Merge pull request #193 from davies/daemon

[SPARKR-214] launch R worker by a daemon

shivaram committed:
Merge pull request #216 from davies/select2

[SPARKR-189] add more selector for DataFrame

davies committed:
address commets

Commit:
Update SQLContext.R

Commit:
Update DataFrame.R

shivaram committed:
Merge pull request #215 from davies/agg

[SPARKR-221] include grouping columns in agg()

davies committed:
Merge branch 'sparkr-sql' of github.com:amplab-extras/SparkR-pkg into toDF

Conflicts:
	pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/SQLUtils.scala

shivaram committed:
Remove staging repo now that 1.3 is released

davies committed:
Merge branch 'master' into merge

Conflicts:
	.gitignore
	pkg/DESCRIPTION
	pkg/NAMESPACE
	pkg/R/RDD.R
	pkg/R/pairRDD.R
	pkg/R/sparkR.R
	pkg/R/utils.R
	pkg/inst/sparkR-submit
	pkg/inst/tests/test_rdd.R
	pkg/inst/worker/worker.R
	pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/RRDD.scala
	pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/SparkRRunner.scala

shivaram committed:
Merge pull request #219 from shivaram/sparkr-build-final

Remove staging repo now that 1.3 is released

shivaram committed:
Merge pull request #179 from evertlammerts/sparkr-sql

[SPARKR-208] Changed SQLContext.R#parquetFile signature to match the java version

shivaram committed:
Include utils before SparkR for `head` to work
Before this change calling `head` on a DataFrame would not
work from the sparkR script as utils would be loaded after SparkR
and placed ahead in the search list. This change requires utils to
be loaded before SparkR

shivaram committed:
Merge pull request #220 from shivaram/sparkr-utils-include

Include utils before SparkR for `head` to work

davies committed:
address comments

Commit:
Update SerDe.scala

davies committed:
handle NULL

davies committed:
handle NULL

shivaram committed:
Merge pull request #212 from davies/toDF

[SPARKR-192]  createDataFrame from rdd

davies committed:
Merge branch 'sparkr-sql' of github.com:amplab-extras/SparkR-pkg into merge

davies committed:
address comments

shivaram committed:
Merge pull request #218 from davies/merge

[SPARKR-225] Merge master into sparkr-sql branch

shivaram committed:
Merge pull request #217 from hlin09/cleanClosureFix

Fix cleanClosure() on recursive function calls.

shivaram committed:
Merge pull request #221 from shivaram/sparkr-stop-start

[SPARKR-219] Clear broadcast variables when JVM is shutdown

davies committed:
add more API for Column

davies committed:
fix test

shivaram committed:
Merge pull request #222 from davies/column2

[SPARKR-233] [SPARKR-190] add more API for Column

Commit:
Moved files to R/

Commit:
Merge branch 'remote_r' into R

Commit:
delete man pages

Commit:
move scala

Commit:
move Scala source into core/ and sql/

Commit:
move R/examples into examples/src/main/r

Commit:
fix path of assembly jar

Commit:
move sparkR into bin/

Commit:
update docs

Commit:
add .gitignore for .o, .so, .Rd

Commit:
add LICENSE header for R sources

Commit:
reduce size of example data

Commit:
fix .gitignore

Commit:
run sparkr tests in Spark

Commit:
rm .travis.yml

Commit:
support R in spark-submit

Commit:
cleanup

SparkQA commented:
[Test build #28747 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28747/consoleFull) for   PR 5077 at commit [`043959e`](https://github.com/apache/spark/commit/043959e7b2d256b9dbf66bca05061223dceb5da0).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28747 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28747/consoleFull) for   PR 5077 at commit [`043959e`](https://github.com/apache/spark/commit/043959e7b2d256b9dbf66bca05061223dceb5da0).
 * This patch **fails RAT tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class SparkRBackend `
  * `class SparkRBackendHandler(server: SparkRBackend)`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28747/
Test FAILed.

Commit:
remove Spark prefix for class

SparkQA commented:
[Test build #28750 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28750/consoleFull) for   PR 5077 at commit [`f7b6936`](https://github.com/apache/spark/commit/f7b69363acf2ca31c41ea2a3e185f707682c20d4).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28750 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28750/consoleFull) for   PR 5077 at commit [`f7b6936`](https://github.com/apache/spark/commit/f7b69363acf2ca31c41ea2a3e185f707682c20d4).
 * This patch **fails RAT tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28750/
Test FAILed.

shivaram committed:
Remove DFC example

shivaram committed:
Fix style

Commit:
fix rat

shivaram committed:
Build SparkR with Maven profile

shivaram committed:
Merge branch 'R' of https://github.com/amplab-extras/spark into R

SparkQA commented:
[Test build #28751 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28751/consoleFull) for   PR 5077 at commit [`716b16f`](https://github.com/apache/spark/commit/716b16f8dfd79eaa60b395b7811ee69aa077f81e).
 * This patch merges cleanly.

shivaram committed:
Add missing comma

SparkQA commented:
[Test build #28752 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28752/consoleFull) for   PR 5077 at commit [`52ca6e5`](https://github.com/apache/spark/commit/52ca6e5a0b2cda693db5b88a958d51660c512918).
 * This patch merges cleanly.

Commit:
change println() to logging

Commit:
Merge branch 'R' of github.com:amplab-extras/spark into R

SparkQA commented:
[Test build #28753 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28753/consoleFull) for   PR 5077 at commit [`9f6aa1f`](https://github.com/apache/spark/commit/9f6aa1f71cf9608a2d5cd9d202218e0703dfa996).
 * This patch merges cleanly.

Commit:
fix bin/sparkR

SparkQA commented:
[Test build #28754 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28754/consoleFull) for   PR 5077 at commit [`0e2412c`](https://github.com/apache/spark/commit/0e2412cee814c567f3760dc33fd44688d5988238).
 * This patch merges cleanly.

Commit:
fix spark-submit with R path and sparkR -h

SparkQA commented:
[Test build #28755 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28755/consoleFull) for   PR 5077 at commit [`ea90fab`](https://github.com/apache/spark/commit/ea90fabf4fae044c8fb9e58589a511bdd0b10aa6).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28751 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28751/consoleFull) for   PR 5077 at commit [`716b16f`](https://github.com/apache/spark/commit/716b16f8dfd79eaa60b395b7811ee69aa077f81e).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28751/
Test PASSed.

SparkQA commented:
[Test build #28752 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28752/consoleFull) for   PR 5077 at commit [`52ca6e5`](https://github.com/apache/spark/commit/52ca6e5a0b2cda693db5b88a958d51660c512918).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28752/
Test PASSed.

SparkQA commented:
[Test build #28753 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28753/consoleFull) for   PR 5077 at commit [`9f6aa1f`](https://github.com/apache/spark/commit/9f6aa1f71cf9608a2d5cd9d202218e0703dfa996).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28753/
Test PASSed.

SparkQA commented:
[Test build #28754 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28754/consoleFull) for   PR 5077 at commit [`0e2412c`](https://github.com/apache/spark/commit/0e2412cee814c567f3760dc33fd44688d5988238).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28754/
Test PASSed.

shivaram committed:
Fix run-tests path

SparkQA commented:
[Test build #28760 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28760/consoleFull) for   PR 5077 at commit [`d6f2bdd`](https://github.com/apache/spark/commit/d6f2bdd28c8829cb1d623a2a8daa4d40ef46386e).
 * This patch merges cleanly.

shivaram committed:
Make bin/sparkR use spark-submit
As a part of this move the R initialization functions
into first.R and first-submit.R

SparkQA commented:
[Test build #28762 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28762/consoleFull) for   PR 5077 at commit [`baefd9e`](https://github.com/apache/spark/commit/baefd9ea6d0eb9424ca61a3c3bdc0631fe438ba5).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28755 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28755/consoleFull) for   PR 5077 at commit [`ea90fab`](https://github.com/apache/spark/commit/ea90fabf4fae044c8fb9e58589a511bdd0b10aa6).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28755/
Test PASSed.

Commit:
fix spark-submit with R scripot

shivaram committed:
Only stop backend JVM if R launched it

SparkQA commented:
[Test build #28766 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28766/consoleFull) for   PR 5077 at commit [`05afef0`](https://github.com/apache/spark/commit/05afef039adb9d6cc9466d8f68b6d5218f528da8).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28760 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28760/consoleFull) for   PR 5077 at commit [`d6f2bdd`](https://github.com/apache/spark/commit/d6f2bdd28c8829cb1d623a2a8daa4d40ef46386e).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28760/
Test PASSed.

SparkQA commented:
[Test build #28762 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28762/consoleFull) for   PR 5077 at commit [`baefd9e`](https://github.com/apache/spark/commit/baefd9ea6d0eb9424ca61a3c3bdc0631fe438ba5).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28762/
Test PASSed.

SparkQA commented:
[Test build #28766 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28766/consoleFull) for   PR 5077 at commit [`05afef0`](https://github.com/apache/spark/commit/05afef039adb9d6cc9466d8f68b6d5218f528da8).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28766/
Test PASSed.

shivaram committed:
Set windows file separators, install dirs

SparkQA commented:
[Test build #28773 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28773/consoleFull) for   PR 5077 at commit [`8030847`](https://github.com/apache/spark/commit/8030847fc611b88b9231b6e6000683783f726742).
 * This patch merges cleanly.

shivaram committed:
Add scripts to start SparkR on windows

shivaram committed:
Merge pull request #214 from sun-rui/SPARKR-156_3

[SPARKR-156] Phase 3: Implement zip() in the RDD class.
Conflicts:
	pkg/inst/worker/worker.R

SparkQA commented:
[Test build #28779 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28779/consoleFull) for   PR 5077 at commit [`cb6e5e3`](https://github.com/apache/spark/commit/cb6e5e3e0fa13123c7a96ad460cf92ef28d18a4d).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28773 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28773/consoleFull) for   PR 5077 at commit [`8030847`](https://github.com/apache/spark/commit/8030847fc611b88b9231b6e6000683783f726742).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28773/
Test PASSed.

SparkQA commented:
[Test build #28779 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28779/consoleFull) for   PR 5077 at commit [`cb6e5e3`](https://github.com/apache/spark/commit/cb6e5e3e0fa13123c7a96ad460cf92ef28d18a4d).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28779/
Test PASSed.

Commit:
Merge branch 'R' of github.com:amplab-extras/spark into R

Conflicts:
	R/pkg/R/sparkRClient.R
	bin/sparkR
	core/src/main/scala/org/apache/spark/deploy/RRunner.scala
	launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

Commit:
Merge branch 'R' of github.com:amplab-extras/spark into R

SparkQA commented:
[Test build #28816 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28816/consoleFull) for   PR 5077 at commit [`42d8b4c`](https://github.com/apache/spark/commit/42d8b4c86a9f4522ccfa8917273bc4e351431915).
 * This patch merges cleanly.

Commit:
fix exit code of sparkr unit test

Commit:
Merge branch 'sparkr-sql' of github.com:amplab-extras/SparkR-pkg into remote_r

Commit:
Merge branch 'remote_r' into R

Commit:
rm man page

Commit:
rm PROJECT_HOME

SparkQA commented:
[Test build #28817 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28817/consoleFull) for   PR 5077 at commit [`56670ef`](https://github.com/apache/spark/commit/56670efe917672c8b14306f4e56ce8d673554366).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28821 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28821/consoleFull) for   PR 5077 at commit [`1a16cd6`](https://github.com/apache/spark/commit/1a16cd6530762b628b513f46dfb043cc68ee6d29).
 * This patch merges cleanly.

Commit:
add missing files

SparkQA commented:
[Test build #28822 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28822/consoleFull) for   PR 5077 at commit [`d436f26`](https://github.com/apache/spark/commit/d436f26264bc99d20d89b6b90e7fc4c0064b6d7a).
 * This patch merges cleanly.

shivaram committed:
Update README remove outdated TODO

Commit:
fix test of zipRDD()

Commit:
Merge branch 'R' of github.com:amplab-extras/spark into R

SparkQA commented:
[Test build #28823 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28823/consoleFull) for   PR 5077 at commit [`ebd4d07`](https://github.com/apache/spark/commit/ebd4d07aa1998b45ace5b41d19d15d9ecc2ed80b).
 * This patch merges cleanly.

Commit:
support R in YARN cluster

SparkQA commented:
[Test build #28824 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28824/consoleFull) for   PR 5077 at commit [`2892e29`](https://github.com/apache/spark/commit/2892e29b4c094eb7deaa3281f8ea6591dcc7a6f9).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28824 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28824/consoleFull) for   PR 5077 at commit [`2892e29`](https://github.com/apache/spark/commit/2892e29b4c094eb7deaa3281f8ea6591dcc7a6f9).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28824/
Test FAILed.

Commit:
fix build

SparkQA commented:
[Test build #28826 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28826/consoleFull) for   PR 5077 at commit [`7da0049`](https://github.com/apache/spark/commit/7da0049425980825f940350933fbae1e163b4540).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28826 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28826/consoleFull) for   PR 5077 at commit [`7da0049`](https://github.com/apache/spark/commit/7da0049425980825f940350933fbae1e163b4540).
 * This patch **fails RAT tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28826/
Test FAILed.

SparkQA commented:
[Test build #28816 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28816/consoleFull) for   PR 5077 at commit [`42d8b4c`](https://github.com/apache/spark/commit/42d8b4c86a9f4522ccfa8917273bc4e351431915).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28816/
Test PASSed.

Commit:
fix license check

SparkQA commented:
[Test build #28827 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28827/consoleFull) for   PR 5077 at commit [`ce3ca62`](https://github.com/apache/spark/commit/ce3ca62af841b90230ee08dc30c0ed27f3c61cea).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28817 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28817/consoleFull) for   PR 5077 at commit [`56670ef`](https://github.com/apache/spark/commit/56670efe917672c8b14306f4e56ce8d673554366).
 * This patch **fails SparkR unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28817/
Test FAILed.

Commit:
disable spark and python tests temporary

SparkQA commented:
[Test build #28829 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28829/consoleFull) for   PR 5077 at commit [`d8b24fc`](https://github.com/apache/spark/commit/d8b24fc7f6f37ab91c04b7c3f6a3dfa9cddeae3d).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28821 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28821/consoleFull) for   PR 5077 at commit [`1a16cd6`](https://github.com/apache/spark/commit/1a16cd6530762b628b513f46dfb043cc68ee6d29).
 * This patch **fails SparkR unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28821/
Test FAILed.

SparkQA commented:
[Test build #28829 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28829/consoleFull) for   PR 5077 at commit [`d8b24fc`](https://github.com/apache/spark/commit/d8b24fc7f6f37ab91c04b7c3f6a3dfa9cddeae3d).
 * This patch **fails SparkR unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28829/
Test FAILed.

SparkQA commented:
[Test build #28823 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28823/consoleFull) for   PR 5077 at commit [`ebd4d07`](https://github.com/apache/spark/commit/ebd4d07aa1998b45ace5b41d19d15d9ecc2ed80b).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28823/
Test FAILed.

SparkQA commented:
[Test build #28822 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28822/consoleFull) for   PR 5077 at commit [`d436f26`](https://github.com/apache/spark/commit/d436f26264bc99d20d89b6b90e7fc4c0064b6d7a).
 * This patch **fails SparkR unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28822/
Test FAILed.

Commit:
fix zipRDD() tests

SparkQA commented:
[Test build #28832 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28832/consoleFull) for   PR 5077 at commit [`410ec18`](https://github.com/apache/spark/commit/410ec184c37bbea5239bc5be07bf96b4add2b992).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28832 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28832/consoleFull) for   PR 5077 at commit [`410ec18`](https://github.com/apache/spark/commit/410ec184c37bbea5239bc5be07bf96b4add2b992).
 * This patch **fails SparkR unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28832/
Test FAILed.

Commit:
fix flaky test

SparkQA commented:
[Test build #28833 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28833/consoleFull) for   PR 5077 at commit [`974e4ea`](https://github.com/apache/spark/commit/974e4eaac85179de9f1b9745574b04ecdf0fccee).
 * This patch merges cleanly.

shivaram committed:
Ignore unknown jobj in cleanup

SparkQA commented:
[Test build #28833 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28833/consoleFull) for   PR 5077 at commit [`974e4ea`](https://github.com/apache/spark/commit/974e4eaac85179de9f1b9745574b04ecdf0fccee).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28833/
Test FAILed.

SparkQA commented:
[Test build #28835 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28835/consoleFull) for   PR 5077 at commit [`423ea3c`](https://github.com/apache/spark/commit/423ea3caae19b2ef7201ae969743dfe715972a65).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28827 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28827/consoleFull) for   PR 5077 at commit [`ce3ca62`](https://github.com/apache/spark/commit/ce3ca62af841b90230ee08dc30c0ed27f3c61cea).
 * This patch **fails SparkR unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28827/
Test FAILed.

Commit:
mark R classes/objects are private

Commit:
Merge branch 'R' of github.com:amplab-extras/spark into R

SparkQA commented:
[Test build #28836 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28836/consoleFull) for   PR 5077 at commit [`855537f`](https://github.com/apache/spark/commit/855537f5f2eb595aed83c80f39148d9aa64cffda).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28835 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28835/consoleFull) for   PR 5077 at commit [`423ea3c`](https://github.com/apache/spark/commit/423ea3caae19b2ef7201ae969743dfe715972a65).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `class RBackend `
  * `class RBackendHandler(server: RBackend)`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28835/
Test FAILed.

davies commented on R/pkg/DESCRIPTION:
```diff
@@ -0,0 +1,35 @@
+Version: 0.1
+Date: 2013-09-09
+Author: Shivaram Venkataraman
+Maintainer: Shivaram Venkataraman <shivaram@cs.berkeley.edu>
```
@shivaram Should we update these fields?

SparkQA commented:
[Test build #28836 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28836/consoleFull) for   PR 5077 at commit [`855537f`](https://github.com/apache/spark/commit/855537f5f2eb595aed83c80f39148d9aa64cffda).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28836/
Test FAILed.

shivaram committed:
Include RStudio instructions in README

SparkQA commented:
[Test build #28837 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28837/consoleFull) for   PR 5077 at commit [`b44e371`](https://github.com/apache/spark/commit/b44e37167000dc5dac31d63b3c840ba461132996).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28837 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28837/consoleFull) for   PR 5077 at commit [`b44e371`](https://github.com/apache/spark/commit/b44e37167000dc5dac31d63b3c840ba461132996).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28837/
Test FAILed.

Commit:
sort generics

Commit:
Merge branch 'R' of github.com:amplab-extras/spark into R

SparkQA commented:
[Test build #28839 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28839/consoleFull) for   PR 5077 at commit [`bdf3a14`](https://github.com/apache/spark/commit/bdf3a149398f2c981411feb74fac873b482d7872).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28839 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28839/consoleFull) for   PR 5077 at commit [`bdf3a14`](https://github.com/apache/spark/commit/bdf3a149398f2c981411feb74fac873b482d7872).
 * This patch **fails SparkR unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28839/
Test FAILed.

shivaram committed:
Fix libPaths in README

SparkQA commented:
[Test build #28840 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28840/consoleFull) for   PR 5077 at commit [`7100fb9`](https://github.com/apache/spark/commit/7100fb9e51bf123713948213a267096cd230d634).
 * This patch merges cleanly.

Commit:
fix flaky tests

Commit:
Merge branch 'R' of github.com:amplab-extras/spark into R

SparkQA commented:
[Test build #28840 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28840/consoleFull) for   PR 5077 at commit [`7100fb9`](https://github.com/apache/spark/commit/7100fb9e51bf123713948213a267096cd230d634).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28840/
Test FAILed.

SparkQA commented:
[Test build #28841 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28841/consoleFull) for   PR 5077 at commit [`afd8a77`](https://github.com/apache/spark/commit/afd8a77d9cc4cd4cbaa8af8e4cfae6babcb16ebb).
 * This patch merges cleanly.

davies commented on dev/run-tests:
```diff
@@ -185,48 +185,58 @@ echo "========================================================================="
-  if [ -n "$_RUN_SQL_TESTS" ]; then
-    SBT_MAVEN_PROFILES_ARGS="$SBT_MAVEN_PROFILES_ARGS -Phive -Phive-thriftserver"
-  fi
+#{
```
Rollback before merge

SparkQA commented:
[Test build #28841 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28841/consoleFull) for   PR 5077 at commit [`afd8a77`](https://github.com/apache/spark/commit/afd8a77d9cc4cd4cbaa8af8e4cfae6babcb16ebb).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28841/
Test FAILed.

Commit:
fix code style

SparkQA commented:
[Test build #28844 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28844/consoleFull) for   PR 5077 at commit [`0e5a83f`](https://github.com/apache/spark/commit/0e5a83f692974752d794d5bc1c9cae17cbc65735).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28844 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28844/consoleFull) for   PR 5077 at commit [`0e5a83f`](https://github.com/apache/spark/commit/0e5a83f692974752d794d5bc1c9cae17cbc65735).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28844/
Test FAILed.

Commit:
enable spark and pyspark tests

davies commented:
@shivaram @pwendell @rxin I think this PR is ready to review.

SparkQA commented:
[Test build #28845 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28845/consoleFull) for   PR 5077 at commit [`d6d3729`](https://github.com/apache/spark/commit/d6d3729dad49655929d3cc3cbee3d55e39e51786).
 * This patch merges cleanly.

shivaram commented:
Thanks @davies  Lets close this PR and I will open a fresh one -- I want to give a better description to the PR and acknowledge all the contributors etc.

shivaram committed:
Add a script to generate R docs (Rd, html)
Also fix some issues with our documentation

shivaram committed:
Merge branch 'R' of https://github.com/amplab-extras/spark into R

SparkQA commented:
[Test build #28847 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28847/consoleFull) for   PR 5077 at commit [`1f478c5`](https://github.com/apache/spark/commit/1f478c5f97179bd8bae56e7277c11b3d6a0fba98).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28847 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28847/consoleFull) for   PR 5077 at commit [`1f478c5`](https://github.com/apache/spark/commit/1f478c5f97179bd8bae56e7277c11b3d6a0fba98).
 * This patch **fails RAT tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28847/
Test FAILed.

shivaram committed:
Add instructions to generate docs

SparkQA commented:
[Test build #28848 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28848/consoleFull) for   PR 5077 at commit [`6ff5ea2`](https://github.com/apache/spark/commit/6ff5ea2d83d84147becdb9c8135f484005f018fe).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28848 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28848/consoleFull) for   PR 5077 at commit [`6ff5ea2`](https://github.com/apache/spark/commit/6ff5ea2d83d84147becdb9c8135f484005f018fe).
 * This patch **fails RAT tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28848/
Test FAILed.

shivaram committed:
Add license to create-docs.sh

shivaram committed:
Merge branch 'R' of https://github.com/amplab-extras/spark into R

SparkQA commented:
[Test build #28851 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28851/consoleFull) for   PR 5077 at commit [`58276f5`](https://github.com/apache/spark/commit/58276f5c5cabdc7175cccc47ee73f6fff9995c92).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28845 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28845/consoleFull) for   PR 5077 at commit [`d6d3729`](https://github.com/apache/spark/commit/d6d3729dad49655929d3cc3cbee3d55e39e51786).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28845/
Test FAILed.

concretevitamin commented:
@shivaram What's the timeline we are looking at? 

If possible I'd like to take a close look at this next week / after next week.

SparkQA commented:
[Test build #28851 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28851/consoleFull) for   PR 5077 at commit [`58276f5`](https://github.com/apache/spark/commit/58276f5c5cabdc7175cccc47ee73f6fff9995c92).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28851/
Test PASSed.

shivaram commented:
@concretevitamin We still have a few PRs open at SparkR-pkg and once they are merged I'll open a fresh PR here. It'll probably be end of this week to get there and after that we'll keep it open here to give enough time for code review.

shivaram committed:
Send Spark INFO logs to a file in SparkR tests

shivaram committed:
Address comments

Commit:
Merge pull request #1 from shivaram/log4j-tests

Send Spark INFO logs to a file in SparkR tests

SparkQA commented:
[Test build #28879 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28879/consoleFull) for   PR 5077 at commit [`b21a0da`](https://github.com/apache/spark/commit/b21a0da098da2aa560094f0177cc9c45ea26ffea).
 * This patch merges cleanly.

Commit:
update R examples (remove master from args)

SparkQA commented:
[Test build #28883 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28883/consoleFull) for   PR 5077 at commit [`733380d`](https://github.com/apache/spark/commit/733380de06bab53294d919c78a248f0f9275f76e).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28879 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28879/consoleFull) for   PR 5077 at commit [`b21a0da`](https://github.com/apache/spark/commit/b21a0da098da2aa560094f0177cc9c45ea26ffea).
 * This patch **fails SparkR unit tests**.

 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28879/
Test FAILed.

Commit:
fix flaky test

SparkQA commented:
[Test build #28885 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28885/consoleFull) for   PR 5077 at commit [`3eacfc0`](https://github.com/apache/spark/commit/3eacfc072758a445d1f01b29001c69683ac5b457).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28883 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28883/consoleFull) for   PR 5077 at commit [`733380d`](https://github.com/apache/spark/commit/733380de06bab53294d919c78a248f0f9275f76e).
 * This patch **passes all tests**.

 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28883/
Test PASSed.

SparkQA commented:
[Test build #28885 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28885/consoleFull) for   PR 5077 at commit [`3eacfc0`](https://github.com/apache/spark/commit/3eacfc072758a445d1f01b29001c69683ac5b457).
 * This patch **passes all tests**.

 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28885/
Test PASSed.

davies commented:
Close this one, will open an new one by @shivaram
